crates/ruma-client-api/src/discovery/get_supported_versions.rs

```diff
@@ -59,37 +59,35 @@ impl Response {
     /// The versions returned will be sorted from oldest to latest. Use [`.find()`][Iterator::find]
     /// or [`.rfind()`][DoubleEndedIterator::rfind] to look for a minimum or maximum version to use
     /// given some constraint.
-    pub fn known_versions(&self) -> impl DoubleEndedIterator<Item = MatrixVersion> {
+    pub fn known_versions(&self) -> BTreeSet<MatrixVersion> {
         self.versions
             .iter()
             // Parse, discard unknown versions
             .flat_map(|s| s.parse::<MatrixVersion>())
-            // Map to key-value pairs where the key is the major-minor representation
-            // (which can be used as a BTreeMap unlike MatrixVersion itself)
-            .map(|v| (v.into_parts(), v))
-            // Collect to BTreeMap
-            .collect::<BTreeMap<_, _>>()
-            // Return an iterator over just the values (`MatrixVersion`s)
-            .into_values()
+            // Collect to BTreeSet
+            .collect()
     }
 }
 
 #[cfg(test)]
 mod tests {
+    use std::collections::BTreeSet;
+
     use ruma_common::api::MatrixVersion;
 
     use super::Response;
 
     #[test]
     fn known_versions() {
         let none = Response::new(vec![]);
-        assert_eq!(none.known_versions().next(), None);
+        assert!(none.known_versions().is_empty());
```

Comment:
Super nitpick, feel free to ignore, but I'd prefer an `assert_eq!(none.known_versions(), BTreeSet::new());` (because it debug-prints the lhs if assertion fails).

Reply:
It's okay to be nitpicky :wink:, I changed it.